pennylane/pytrees.py

```diff
@@ -54,5 +105,93 @@ def register_pytree(pytree_type: type, flatten_fn: FlattenFn, unflatten_fn: Unfl
 
     """
 
+    _register_pytree_with_pennylane(pytree_type, flatten_fn, unflatten_fn)
+
     if has_jax:
         _register_pytree_with_jax(pytree_type, flatten_fn, unflatten_fn)
+
+
+class Structure(namedtuple("Structure", ["type", "metadata", "children"])):
+    """A pytree data structure, holding the type, metadata, and child pytree structures."""
+
+    def __repr__(self):
+        return f"PyTree({self.type.__name__}, {self.metadata}, {self.children})"
```

Comment:
```suggestion
@dataclass(repr=False)
class Structure:
    """A pytree data structure, holding the type, metadata, and child pytree structures."""
    
    type: type
    metadata: Metadata
    children: list[Leaf, 'Structure']
    
    def __repr__(self):
        return f"PyTree({self.type.__name__}, {self.metadata}, {self.children})"
```
I think dataclass would work better here- better type hints and it integrates with `orjson` for serialization